src/test/java/io/airlift/slice/TestSliceUtf8.java

```diff
@@ -134,6 +136,12 @@
         INVALID_SEQUENCES = invalidSequences.build();
     }
 
+    private static final int[] UTF_16_BOUNARY_CODE_POINTS = new int[] {0x0000, 0xD7FF, 0xE000, 0xFFFF, 0x10000, 0x10FFFF};
+    private static final int[] UTF_16_INTERESTING_CODE_POINTS = Arrays.copyOf(ALL_CODE_POINTS, 500);
```

Comment:
Actually, the chance of picking anything in the range of E000 and FFFF is rather low. It would be better if the 3 ranges are sampled separately.